### PR TITLE
Revise the appendix on exception

### PR DESCRIPTION
**Pull request by jbarnoud**
I did quite extensive changes in appendix E. I'd like somebody to have a look before merging them.

### CODE REVIEW

**Review by pckroon — APPROVED** [review]

File: source/PvL_E.rst [hunk 1]
```diff
@@ -55,85 +55,77 @@ error before the colon, and specifics about the error after the colon.
...
+            # This will be executed if not exception got raised in the
+            # ``try`` statement.
+            print('The square of your number is ', user_input_as_number**2)
+        finally:
+            # This will be executed whether or not an exception is raised.
+            print('Thank you')
+
+When using a ``try`` clause, you should have as little as possible in the
+``try`` block. If too many things happen in that block, you take the risk
+handle an unexpected exception.
```
Comment (pckroon) [inline source/PvL_E.rst]: ... you take the risk *of handling* an unexpected exception
or
... you risk handling an unexpected exception

File: source/PvL_E.rst [hunk 2]
```diff
@@ -55,85 +55,77 @@ error before the colon, and specifics about the error after the colon.
...
+        try:
+            with open(filename) as infile:
+                content = infile.read()
+        except FileNotFoundError:
+            print('The file does not exist.')
+        except PermissionError:
+            print('Your are not allowed to read this file.')
+
+It is also possible not to specify a particular exception in the ``except``
+statement. In thy case, any exception will be handled. Such bare ``except``
```
Comment (pckroon) [inline source/PvL_E.rst]: this case

File: source/PvL_E.rst [hunk 3]
```diff
@@ -302,6 +294,4 @@ Exercises
    integer and then checks the input to confirm that it meets the requirements. 
    It should be able to handle inputs that cannot be converted to ``int``, as well
    as negative ``int``\s, and edge cases (e.g. when the user closes the dialog, or
-   does not enter anything at all.)   
-   
-
+   does not enter anything at all.)
```
Comment (pckroon) [inline source/PvL_E.rst:297]: Please keep a newline at the end of the file

**Review by pckroon — CHANGES_REQUESTED** [review]
Good in general, some small typos

File: source/PvL_E.rst (hunk 3 above)
Comment (jbarnoud) [reply source/PvL_E.rst:297]: There is one.

**Review by pckroon — APPROVED** [review]

**Pull request merged**